### PR TITLE
Handle failures onPacket

### PR DESCRIPTION
**Pull request by geiralund**
- Services can override error handling (problem) by implementing onFailure method.

### CODE REVIEW

File: src/main/kotlin/no/nav/dagpenger/streams/River.kt [hunk 1]
```diff
@@ -26,4 +40,12 @@ abstract class River : Service() {
 
     abstract fun filterPredicates(): List<Predicate<String, Packet>>
     abstract fun onPacket(packet: Packet): Packet
-}
+    open fun onFailure(packet: Packet): Packet {
+        packet.addProblem(
+            Problem(
```
Comment (geiralund) [inline src/main/kotlin/no/nav/dagpenger/streams/River.kt]: we may want to specify a specific type (ie urn:dp:error.. )

File: src/main/kotlin/no/nav/dagpenger/streams/River.kt [hunk 2]
```diff
@@ -14,10 +15,23 @@ abstract class River : Service() {
...
+                }
+                packet.hasProblem()
+            }
             .filter { key, packet -> filterPredicates().all { it.test(key, packet) } }
             .mapValues { _, packet ->
-                onPacket(packet)
+                val result = runCatching { onPacket(packet) }
+                return@mapValues when {
+                    result.isFailure -> {
+                        LOGGER.error(result.exceptionOrNull()) { "Failed to process packet" }
```
Comment (geiralund) [inline src/main/kotlin/no/nav/dagpenger/streams/River.kt]: add a prometheus error counter here?

File: src/main/kotlin/no/nav/dagpenger/streams/River.kt [hunk 3]
```diff
@@ -14,10 +15,23 @@ abstract class River : Service() {
     override fun buildTopology(): Topology {
         val builder = StreamsBuilder()
         val stream = builder.consumeTopic(DAGPENGER_BEHOV_PACKET_EVENT)
-        stream.peek { key, packet -> LOGGER.info("Processing $packet with key $key") }
+        stream
```
Comment (gtcno) [inline src/main/kotlin/no/nav/dagpenger/streams/River.kt]: Should you care if this is a packet you are not interested in?

Comment (geiralund) [reply src/main/kotlin/no/nav/dagpenger/streams/River.kt]: no, good point. Will remove it - already log an error on runCatching

**Pull request merged**